tests/core/modules/users/resources.py

```diff
@@ -0,0 +1,43 @@
+from superdesk.core.mongo import MongoResourceConfig, MongoIndexOptions
+from superdesk.core.elastic.resources import ElasticResourceConfig
+
+from superdesk.core.resources import ResourceModelConfig
+from superdesk.core.resources.service import AsyncResourceService
+from superdesk.core.http.resource_endpoints import ResourceEndpoints
+
+
+from .types import User
+
+
+class UserResourceService(AsyncResourceService[User]):
+    resource_name = "users_async"
+
+
+user_model_config = ResourceModelConfig(
+    name="users_async",
+    data_class=User,
+    mongo=MongoResourceConfig(
+        indexes=[
+            MongoIndexOptions(
+                name="users_name_1",
+                keys=[("first_name", 1)],
+            ),
+            MongoIndexOptions(
+                name="combined_name_1",
+                keys=[("first_name", 1), ("last_name", -1)],
+                background=False,
+                unique=False,
+                sparse=False,
+                collation={"locale": "en", "strength": 1},
+            ),
+        ],
+    ),
+    elastic=ElasticResourceConfig(),
+    service=UserResourceService,
+)
+
+users_resource_endpoints = ResourceEndpoints(
```

Comment:
This one should be limited to urls/web API.
I've separated out the data (service) and http (endpoint), so we have a clear distinction as to what's coming from the API and what is being called internally

Reply:
Another example here, if a resource is not to be exposed via the API, then this endpoint class is never used for that resource

Reply:
yep I'm not against that being configurable, just not sure if we need to create another class instance just to configure it (which will be the default for most resources) instead of having that config in the resource model

Reply:
Moved this onto the `ResourceConfig` (also renamed from `ResourceModelConfig`, as the config has more than just model config).